js/src/files/write.js

```diff
@@ -48,5 +48,14 @@ module.exports = (createCommon, options) => {
         done()
       })
     })
+
+    it('should write to deeply nested non existent file with create and parents flags, expect no error', function (done) {
```

Comment:
This test doesn't actually test if a parent directory is being created, only that the function doesn't error

Reply:
That's true.  Neither do the other tests present, nor do the related `go-ipfs` sharness tests.

Reply:
😢

Reply:
Looks better now though! Thanks for fixing it + the other test